web/web.go

```diff
@@ -65,8 +65,17 @@ type Handler struct {
 	reloadCh    chan struct{}
 	options     *Options
 	statusInfo  *PrometheusStatus
+}
+
+// ApplyConfig updates the status state as the new config requires.
+// Returns true on success.
+func (h *Handler) ApplyConfig(conf *config.Config) bool {
+	h.federation.mtx.Lock()
```

Comment:
I was confused when reading the code at first because I couldn't find any `Lock()` usage in federate.go, only `RLock()`. Maybe have any `ApplyConfig()` defined on `Federation` to have all its lock handling within the type?